### PR TITLE
Fail on deprecation warnings in tooling api cross version tests

### PR DESCRIPTION
**Pull request by reinsch82**

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test ACT

File: subprojects/kotlin-dsl-tooling-builders/src/integTest/groovy/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelSpec.groovy [hunk 1]
```diff
@@ -16,48 +16,46 @@
...
+        Map<File, KotlinDslScriptsModel> singleRequestModels = model.scriptModels
 
         then:
 
         singleRequestModels[spec.scripts["a"]].exceptions.size() == 1
         singleRequestModels[spec.scripts["b"]].exceptions.size() == 1
         singleRequestModels[spec.scripts["settings"]].exceptions.isEmpty()
-    }
 
+        validateOutput()
```
Comment (eskatos) [inline subprojects/kotlin-dsl-tooling-builders/src/integTest/groovy/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelSpec.groovy:59]: :question: is it required to call `validateOutput()` in each test in order to fail on deprecation warnings? I would have expected this is automatic in all cross version tests and we add explicit expectations to make the tests pass.

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Pull request closed without merging**